### DISCUSSION TITLE
Welcome to node-red-contrib-myhome-bticino-v2 Discussions!

### DISCUSSION THREAD

**Discussion by FredBlo** [FredBlo/node-red-contrib-myhome-bticino-v2#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chassotce** [comment]


**Reply by FredBlo** [reply]
Hi @chassotce ,

The question is actually whether a node-red connector exists for Home Asssistant...
But since Home Assistant is an integration tool by itself, not sure it does not become a de-duplicated solution... (but it workarounds not having such connector available on Home Assistant...)

Already tried this ?
https://flows.nodered.org/node/node-red-contrib-ha-mqtt

**Comment by chassotce** [comment]
Thanks for the answer, i tried your plugin and i cant connect to my
homeserver1, Im a little stuck, if you want try to help me (discord maybe)
(i speak french). Thanks you

**Reply by FredBlo** [reply]
Globally, I like the idea on OpenSource projects where problems one may encounter are shared, all are able to see and answer when they have the knowledge or experience, so that it build up the community experience, for all other users, potentially encountering the same kind of issue or challenge.

Can you describe what you tried, the issues you encountered, some console/logs info...
If this is as specific issue or bug, you can also open an issue on GitHub

Thanks !

**Reply by chassotce** [reply]
I have try to connect with the open password of myHomeServer_1 but i have this error : 
"gateway connection issue (socket connection closed): last known state was 'disconnected', trying to re-connect..."

I have see for myHomeServer i have to use HMAC but i don't know what to put in password input of the gateway, can you help me please ?

I have pass 3 days trying, changing password , read topics ... nothing help

**Reply by FredBlo** [reply]
Hi,

I am wondering whether your are using the right password. You have to use the 'OPEN' password.

MyHomeServer1 has different access levels and passwords (installer, admin, user(s), OPEN).
Did you define the OPEN one in the app (in **Installer mode**) to enable external (add-in) use ?
![image](https://user-images.githubusercontent.com/78979088/142631801-2776fd90-ccb2-4a4b-8cb2-e7900513678e.png)  ![image](https://user-images.githubusercontent.com/78979088/142631729-8f6ece3f-5f1f-4157-a79d-23bb323b59a6.png)

**Reply by chassotce** [reply]
Yes, I did. I have change to a easy one to test and the same .... and i can connect with myhome_suite , it use the port 20000 and the open password, maybe we can see it together in live and when we found, we put a topic on there ?

**Comment by FredBlo** [comment]
Hi @chassotce ,

I am not the live support of BTicino / Legrand... :-P
I can help you by providing what I would try, aftwerwards its your try...

Since you mentioned quite very few info about what you attempted, no idea how far you tried before by yourself.
Have you tried :
- to basically connect to the gateway using the standard Legrand tool MyHome Suite (from your PC). It also connects using OPEN protocol [download from here](https://www.homesystems-legrandgroup.com/home?p_p_id=it_smc_bticino_homesystems_search_AutocompletesearchPortlet&p_p_lifecycle=0&p_p_state=normal&p_p_mode=view&_it_smc_bticino_homesystems_search_AutocompletesearchPortlet_journalArticleId=2493426&_it_smc_bticino_homesystems_search_AutocompletesearchPortlet_mvcPath=%2Fview_journal_article_content.jsp)
- to connect using the (non user-friendly) OPEN tool from Bticino [download from here](https://developer.legrand.com/documentation/client-download/)
- completely disconnect your node-red server and wait > 1 hour before retrying (your IP may have been banned by the gateway if it tried too many times with an incorrect password)
- collect info from the node-red server console : more info on there (about password incorrect or something else,...)

**Reply by chassotce** [reply]
Hi,

Thanks for the answer:

- works
- works
- have to try
- only have close by port
